src/Components/Server/src/Circuits/ServerComponentDeserializer.cs

```diff
@@ -0,0 +1,254 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using Microsoft.AspNetCore.DataProtection;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Components.Server
+{
+    // Shared settings for serializing and deserializing server components
+    // **Component descriptor protocol**
+    // MVC serializes one or more components as comments in HTML.
+    // Each comment is in the form <!-- Blazor: <<Json>> -->
```

Comment:
nitpick, include an example of the json... `{ type: ..., sequence:... }` and so on.

Reply:
Added a sample